### PR TITLE
batch_coprocessor: change config for batch_coprocessor's timeout

### PR DESCRIPTION
**Pull request by hanfei1991**
### What problem does this PR solve?

- TiFlash always do heavy computation, so `timeout` shoule be large enough. This PR set it 60min, which can cover most cases.
- Secondly, when timeout happens, tiflash should not retry, because it must be a logic error but not eviroument error (like io error)

### What is changed and how it works?

What's Changed:
Timeout config is larger and on send fail will not retry for `canceled error`

Tests 

Test is implemented in TiFlash Side

### Release note 

- fix bugs of `tikv server timeout` when tidb_allow_batch_cop is enabled.

### CODE REVIEW

File: store/tikv/client.go [hunk 1]
```diff
@@ -50,9 +50,10 @@ var MaxRecvMsgSize = math.MaxInt64
 // Timeout durations.
 var (
 	dialTimeout               = 5 * time.Second
-	readTimeoutShort          = 20 * time.Second  // For requests that read/write several key-values.
-	ReadTimeoutMedium         = 60 * time.Second  // For requests that may need scan region.
-	ReadTimeoutLong           = 150 * time.Second // For requests that may need scan region multiple times.
+	readTimeoutShort          = 20 * time.Second   // For requests that read/write several key-values.
+	ReadTimeoutMedium         = 60 * time.Second   // For requests that may need scan region.
+	ReadTimeoutLong           = 150 * time.Second  // For requests that may need scan region multiple times.
+	ReadTimeoutTiFlash        = 3600 * time.Second // For requests that may scan many regions for tiflash.
```
Comment (ilovesoup) [inline store/tikv/client.go:56]: Any chance we don't hardcode TiFlash here? I suggest use things like ultra long.

**Comment by lzmhhh123** [comment]
Should we cherry-pick this PR to release-3.1?

**Comment by hanfei1991** [comment]
@lzmhhh123 It seems release 3.1 doesn't have bath_cop.

**Pull request merged**